src/server/coprocessor/executor/table_scan.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use server::coprocessor::endpoint::{prefix_next, is_point};
+use server::coprocessor::Result;
+use tipb::executor::TableScan;
+use kvproto::coprocessor::KeyRange;
+use storage::{SnapshotStore, Statistics};
+use util::codec::table;
+use util::collections::HashSet;
+use super::{Executor, Row};
+use super::base_scanner::BaseScanner;
+
+struct TableScanExec<'a> {
+    meta: TableScan,
+    col_ids: HashSet<i64>,
+    cursor: usize,
+    key_ranges: Vec<KeyRange>,
+    scanner: BaseScanner<'a>,
+}
+
+impl<'a> TableScanExec<'a> {
+    #[allow(dead_code)] //TODO:remove it
```

Comment:
Since we do not use `TableScanExec::new()` in the code now, and without it would cause warnings in compile, this may also make the `CI` failed. @hhkbp2